Lib/idlelib/config.py

```diff
@@ -778,9 +779,99 @@ def _warn(msg, *key):
         _warned.add(key)
 
 
+class ConfigChanges(dict):
+    """Manage a user's proposed configuration option changes.
+
+    Names used across multiple methods:
+        page -- one of the 4 top-level dicts representing a
+                .idlerc/config-x.cfg file.
+        config_type -- name of a page.
+        section -- a section within a page/file.
+        item -- name of a value within a section.
+        value -- value for the item.
+
+    Methods
+        add_item:
+        save_all: Save all the changes to the config file.
+
+        reset: Clear all changes by clearing each page.
+        set_user_value: Set value *in idleConf* for page, section, item.
+    """
+    def __init__(self):
+        "Create a page for each configuration file"
+        self.pages = []  # List of unhashable dicts.
+        for config_type in ('main', 'highlight', 'keys', 'extensions'):
+            self[config_type] = {}
+            self.pages.append(self[config_type])
+
+    def additem(self, config_type, section, item, value):
+        "Add item/value pair for config_type and section."
+        page = self[config_type]
+        value = str(value)  # Make sure we use a string.
+        if section not in page:
+            page[section] = {}
+        page[section][item] = value
+
+    @staticmethod
+    def set_value(config_type, section, item, value):
```

Comment:
save_option for consistency with add_option and save_all ;-).  Only called in save_all.